Sources/Quick/QuickSpec.swift

```diff
@@ -68,25 +68,30 @@ open class QuickSpec: QuickSpecBase {
     }
 
     private static func addInstanceMethod(for example: Example, classSelectorNames selectorNames : inout Set<String>) -> Selector {
-        let block: @convention(block) (QuickSpec) -> Void = { spec in
+        let block: @convention(block) (QuickSpec, @escaping () -> Void) -> Void = { spec, completionHandler in
             spec.example = example
-            example.run()
+            Task {
+                await example.run()
+                completionHandler()
+            }
         }
         let implementation = imp_implementationWithBlock(block as Any)
 
         let originalName = example.name.c99ExtendedIdentifier
         var selectorName = originalName
         var index: UInt = 2
 
-        while selectorNames.contains(selectorName) {
+        while selectorNames.contains(selectorName.appending(":")) {
             selectorName = String(format: "%@_%tu", originalName, index)
             index += 1
         }
 
+        selectorName += ":"
+
         selectorNames.insert(selectorName)
 
         let selector = NSSelectorFromString(selectorName)
-        class_addMethod(self, selector, implementation, "v@:")
+        class_addMethod(self, selector, implementation, "v@:@?<v@?>")
```

Comment:
Similarly to what we do in QuickSpec.m below, we need to also pass in the correct typing information for the completion handler method.